sympy/logic/boolalg.py

```diff
@@ -179,6 +179,10 @@ def binary_symbols(self):
                            if i.is_Boolean or i.is_Symbol
                            or isinstance(i, (Eq, Ne))])
 
+    def _eval_refine(self, assumptions):
+        from sympy.assumptions import ask
+        return ask(self, assumptions)
```

Comment:
At least it should return the expression unchanged if `ask` returns None. refine is supposed to simplify the input, but `None` is not a simplification of a boolean expression.

Reply:
That is already handled by `refine`.

```python
>>> print(ask(Q.even(x), Q.real(x)))
None
>>> refine(Q.even(x), Q.real(x))
Q.even(x)
```